libtransmission/crypto-utils.h

```diff
@@ -90,6 +90,7 @@ void tr_x509_cert_free(tr_x509_cert_t handle);
  * @brief Returns a random number in the range of [0...upper_bound).
  */
 [[nodiscard]] int tr_rand_int(int upper_bound);
+[[nodiscard]] size_t tr_rand_int(size_t upper_bound);
```

Comment:
Are two variants needed?

If so, should probably be a template function and either move the implementation into the header (see:  tr_rand_obj()) or declare it a template in the header and instantiate it in the cc file (see: tr_parseNum<> in utils.h and utils.cc

Reply:
As I wrote in the PR description, my attempts to use template were all failures:
- Undefined Symbol
- Function cannot be defined in an explicit instantiation
- Function template partial specialization is not allowed
- ...

I was not able to find anything that compiles.

Reply:
I tried for an hour: I'm still unsuccessful at figuring out a construction that compiles.